pkg/route/api/v1/types.go

```diff
@@ -48,6 +48,13 @@ type RouteSpec struct {
 
 	// TLS provides the ability to configure certificates and termination for the route
 	TLS *TLSConfig `json:"tls,omitempty" description:"provides the ability to configure certificates and termination for the route"`
+
+	// RouterShard has information of a routing shard and is used to
+	// generate host names
+	Shard *RouterShard `json:"shard,omitempty" description:"provides shard information for the route"`
```

Comment:
I think @smarterclayton's vision for this was to have it in the status, with the possibility of multiple shards satisfying a single route